aea/aea.py

```diff
@@ -61,9 +62,20 @@ def __init__(self, name: str,
         self._directory = directory if directory else str(Path(".").absolute())
 
         self.mailbox = mailbox
-        self._context = AgentContext(self.name, self.crypto.public_key, self.outbox)
+        self._decision_maker = DecisionMaker(self.max_reactions, self.outbox)
+        self._context = AgentContext(self.name,
+                                     self.crypto.public_key,
+                                     self.outbox,
+                                     self.decision_maker.message_queue,
+                                     self.decision_maker.ownership_state,
```

Comment:
So we distinguish now between preferences and ownership spate. That is a relevant separation but might not be generic enough. We need to see.